ozaria/site/views/play/level/tome/SpellView.coffee

```diff
@@ -833,49 +840,73 @@ module.exports = class SpellView extends CocoView
   # * The editor will reserve space for one annotation as a codeless area.
   # - Problem alerts and ranges will only show on fully cast worlds. Annotations will show continually.
 
+  fetchToken: (source, language) =>
+    if language not in ['java', 'cpp']
+      return Promise.resolve(source)
+    else if source of @loadedToken
+      return Promise.resolve(@loadedToken[source])
+
+    headers =  { 'Accept': 'application/json', 'Content-Type': 'application/json' }
+    m = document.cookie.match(/JWT=([a-zA-Z0-9.]+)/)
+    service = window?.localStorage?.kodeKeeperService or "https://asm14w94nk.execute-api.us-east-1.amazonaws.com/service/parse-code-kodekeeper"
+    if me.useChinaServices()
+      headers['Authorization'] = 'APPCODE b3e285d032a343db8bd2b51a05a5ff1d'
```

Comment:
ok with exposing this in public?

Reply:
🤔 it is public in china branch for a long time, but maybe not in master ? i can clean this code